### PR TITLE
Mounting succeeds gcsbucket doesnot exist

### PR DESCRIPTION
**Pull request by Tulsishah**
Tried to solve an error Mounting succeeds even if gcs call fails.

### CODE REVIEW

File: vendor/github.com/jacobsa/gcloud/gcs/conn.go [hunk 1]
```diff
@@ -184,20 +185,32 @@ func (c *conn) OpenBucket(
 	// bucket.
 	_, err = b.ListObjects(ctx, &ListObjectsRequest{MaxResults: 1})
 
-	if typed, ok := err.(*googleapi.Error); ok {
-		switch typed.Code {
-		case http.StatusForbidden:
+	//if typed, ok := err.(*googleapi.Error); ok {
```
Comment (vadlakondaswetha) [inline vendor/github.com/jacobsa/gcloud/gcs/conn.go:188]: please dont raise cls with commented code.

File: vendor/github.com/jacobsa/gcloud/gcs/conn.go [hunk 2]
```diff
@@ -184,20 +185,32 @@ func (c *conn) OpenBucket(
...
+	//		return
+	//
+	//	case http.StatusNotFound:
+	//		err = fmt.Errorf("Unknown bucket %q", b.Name())
+	//		return
+	//	}
+	//}
+
+	if err != nil {
+		if strings.Contains(err.Error(), "Error 403") {
```
Comment (vadlakondaswetha) [inline vendor/github.com/jacobsa/gcloud/gcs/conn.go:205]: as discussed please dont use string comparision for error. cast it appropriately like this and compare the status code.
(err.(*fmt.wrapError).err).(*googleapi.Error).Code

File: vendor/github.com/jacobsa/gcloud/gcs/conn.go [hunk 3]
```diff
@@ -20,15 +20,16 @@ import (
 	"log"
 	"net/http"
 	"net/url"
+	"strings"
```
Comment (vadlakondaswetha) [inline vendor/github.com/jacobsa/gcloud/gcs/conn.go:23]: add unit tests

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]
Please make changes in jacobsa/gcloud repo

**Comment by raj-prince** [comment]
Just for the information, this bug is similar to the issue open in the gcloud codebase: https://github.com/jacobsa/gcloud/issues/15

File: vendor/github.com/jacobsa/gcloud/gcs/conn.go (hunk 2 above)
Comment (vadlakondaswetha) [reply vendor/github.com/jacobsa/gcloud/gcs/conn.go:205]: Also check if we can cast error as mentioned in the link Prince shared: https://github.com/jacobsa/gcloud/issues/15

**Comment by Tulsishah** [comment]
raise the same PR on jacobsa/gcloud repo.

**Pull request closed without merging**